tests/integration_test_net.go

```diff
@@ -169,6 +177,66 @@ type integrationTestNode struct {
 	done      <-chan struct{}
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Memory profiler.
+// if enabled with the `-test.heap.profile` flag, it will write a heap dump to
+// the `../build/profile/` directory at the end of the test run.
+// The file will be named `mem_<test_name>.pprof` where `<test_name>` is the name
+// of the test that started the profiling.
+////////////////////////////////////////////////////////////////////////////////
+
+type netProfiler struct {
```

Comment:
nit: since it is collecting memory data and not network statistics, should this be renamed to `memoryProfiler`? The fact that it is used by the network should be of lesser importance. Wdyt?

Reply:
struct removed.